src/flb_hash.c

```diff
@@ -77,6 +77,21 @@ struct flb_hash *flb_hash_create(int evict_mode, size_t size, int max_entries)
     return ht;
 }
 
+struct flb_hash *flb_hash_create_with_ttl(int cache_ttl, int evict_mode, 
+                                          size_t size, int max_entries)
+{
+    struct flb_hash *ht;
+
+    ht = flb_hash_create(evict_mode, size, max_entries);
+    if (!ht) {
+        flb_errno();
+        return NULL;
+    }
+
+    ht->cache_ttl = cache_ttl;
```

Comment:
I think you need to initialize the ttl to be 0 or -1 in the old `flb_hash_create` function. 

Right now I see the flb_hash struct allocated like this:

```
ht = flb_malloc(sizeof(struct flb_hash));
```

Malloc will not initialize memory, so the values will be random. To initialize memory to zero, you have to use `flb_calloc`.